testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergProcedureCalls.java

```diff
@@ -111,6 +115,46 @@ public void testMigrateHiveTableWithSmallintType(String fileFormat)
         onTrino().executeQuery("DROP TABLE " + icebergTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "fileFormats")
+    public void testMigrateHiveTableWithComplexType(String fileFormat)
+    {
+        String tableName = "test_migrate_complex_" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+        String icebergTableName = "iceberg.default." + tableName;
+        String sparkTableName = "iceberg_test.default." + tableName;
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " WITH (format='" + fileFormat + "') AS " +
+                "SELECT 1 x, array[2, 3] a, " +
+                "CAST(map(array['key'], array['value']) AS map(varchar, varchar)) b, " +
+                "CAST(row(1) AS row(d integer)) c");
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + icebergTableName))
+                .satisfies(queryResult -> {
+                    assertThat(queryResult).hasRowsCount(1).hasColumnsCount(4);
+                    List<?> row = queryResult.row(0);
+
+                    assertThat(row.get(0)).isEqualTo(1);
+                    assertThat(row.get(1)).isInstanceOf(TrinoArray.class);
+                    assertThat(((TrinoArray) row.get(1)).getArray()).isEqualTo(new Object[] {2, 3});
+                    assertThat(row.get(2)).isEqualTo(ImmutableMap.of("key", "value"));
```

Comment:
We could simply retrieve elements with subscript operators like `SELECT x, a[0], ...`. That's how we verify complex types in other existing product tests.